pkg/solana/logpoller/filters.go

```
@@ -118,6 +140,21 @@ func (fl *filters) RegisterFilter(ctx context.Context, filter Filter) error {
 	if !filter.IsBackfilled {
 		fl.filtersToBackfill[filter.ID] = struct{}{}
 	}
+
+	programID := filter.Address.ToSolana().String()
+	if _, ok = fl.knownPrograms[programID]; !ok {
+		fl.knownPrograms[programID] = 1
+	} else {
+		fl.knownPrograms[programID]++
+	}
+
+	discriminatorHead := filter.Discriminator()[:10]
```

Are we sure that Distriminator has at least 10 bytes? Should we add validation?

---

The `utils.Discriminator` explicitly returns [8]byte, and the `filter.Discriminator` wrapper around that just calls `base64.EncodeToString` on that which will always return []byte backed by [12]byte array. 

I can make this a bit more explicit by changing the return type of `filter.Discriminator` to `[12]byte`, since we have to generate a new slice of it anyway (at least in this case).

---

I forgot that the return type is `string` not `[]byte`. So instead of changing the return type, I added an AssumptionViolation check that panics if the base64 encoder doesn't encode the 8 bytes passed to it as a 12-byte base64 string. I think the panic makes sense here, because the only way this could potentially happen is during development, if we're updating to a new version of `go` and they've broken backwards compatibility in `encoder/base64` or introduced a bug.  The panic will let us know immediately that we need to roll back or update our code accordingly.